lib/alba/resource.rb

```diff
@@ -156,15 +157,16 @@ def transform_key(key)
         @_transform_key_function.call(key.to_s)
       end
 
-      def fetch_attribute(object, attribute)
-        case attribute
-        when Symbol then object.public_send attribute
-        when Proc then instance_exec(object, &attribute)
-        when Alba::One, Alba::Many then yield_if_within(attribute.name.to_sym) { |within| attribute.to_hash(object, params: params, within: within) }
-        when TypedAttribute then attribute.value(object)
-        else
-          raise ::Alba::Error, "Unsupported type of attribute: #{attribute.class}"
-        end
+      def fetch_attribute(object, key, attribute)
+        value = case attribute
+                when Symbol then object.public_send attribute
+                when Proc then instance_exec(object, &attribute)
+                when Alba::One, Alba::Many then yield_if_within(attribute.name.to_sym) { |within| attribute.to_hash(object, params: params, within: within) }
+                when TypedAttribute then attribute.value(object)
+                else
+                  raise ::Alba::Error, "Unsupported type of attribute: #{attribute.class}"
+                end
+        value.nil? && @_on_nil ? instance_exec(object, key, attribute, &@_on_nil) : value
```

Comment:
`instance_exec` will run a potential block given when the value is nil ?

Reply:
When `value` is `nil`, `on_nil` block is executed.